solr/test-framework/src/java/org/apache/solr/SolrTestCaseHS.java

```diff
@@ -190,9 +194,31 @@ public static String getQueryResponse(String wt, SolrParams params) throws Excep
     p.remove("qt");
     p.set("indent", "true");
 
-    DirectSolrConnection connection = new DirectSolrConnection(h.getCore());
-    String raw = connection.request(path, p, null);
-    return raw;
+    SolrCore core = h.getCore();
```

Comment:
I see a bunch of code added here that is inside-Solr layer code instead of client layer.  LocalSolrQueryRequest is inside-Solr layer and lots of ancient tests use it... I see you adding another use here.  I'm aware lots of STCJ4 tests effectively use that indirectly but at least its invisible plumbing to the test.  Can't we use EmbeddedSolrServer, and InputStreamResponseParser to a string... XML/JSON as desired?

(this comment applies to basically everywhere in your PR that you use LocalSolrQueryRequest)

Reply:
I was able to eliminate the LocalSolrQueryRequest use in TestHarness.   I'm now taking a stab at SolrTestCaseHS.